File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO1/BridgePattern/BridgePattern/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace BridgePattern
{
    public abstract class Shape
    {
        public static void ParseLambdas(string text, params Action<float>[] pl)
        {
            string[] tb = text.Split(" \t;".ToArray(), StringSplitOptions.RemoveEmptyEntries);
            int size = tb.Length;
            if (size != pl.Length) throw new ArgumentException("Unexpected prameters count");
            for (int i = 0; i < size; ++i) pl[i](float.Parse(tb[i]));
        }
        public abstract void Draw(Graphics g);
        public abstract string ToXml();
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022s/IZ08IO3/BridgeTest/src/Shape.java

import java.awt.Graphics;

public abstract class Shape
{
   protected int[] tb;
   public Shape(String text,int count)
   {
      String[] params=text.split("[\\s\t;]+");
      if(params.length!=count) throw new IllegalArgumentException("Niueodpowiednia ilość parametrów");
      tb=new int[count];
      for(int i=0;i<count;++i) tb[i]=Integer.parseInt(params[i]);
   }
   public abstract void Draw(Graphics g);
   public abstract String toXml();
}
